dcc-portal-ui/app/scripts/oncogrid/js/controller.js

```diff
@@ -113,4 +113,157 @@
         };
 
     });
+
+    module.controller('OncoGridUploadController', function(Restangular, SetNameService, donorsLimit, genesLimit, filters, 
+        LocationService, $scope, $modalInstance, $location, $q, SetService, $timeout){
+
+        $scope.donorsLimit = donorsLimit;
+        $scope.genesLimit = genesLimit;
+        $scope.filters = filters;
+        $scope.isLaunchingOncoGrid = false;
+
+        $scope.maxDonorsLimit = 3000;
+        $scope.maxGenesLimit = 100;
+
+        $scope.params = {};
+        $scope.hasValidParams= false;
+
+        $scope.params.donorsCount = Math.min($scope.donorsLimit || 3000, $scope.maxDonorsLimit);
+        $scope.params.genesCount = Math.min($scope.genesLimit || 100, $scope.maxGenesLimit);
+        $scope.params.setName = '';
+
+        $scope.hasValidDonorCount = function(value){
+            var val = parseInt(value,10);
+            if (isNaN(val)) {
+                return false;
+            }
+            if (!angular.isNumber(val) || val > $scope.maxDonorsLimit || val <= 0 || val > $scope.donorsLimit) {
+                return false;
+            }
+            return true;
+        }
+
+        $scope.hasValidGeneCount = function(value){
+            var val = parseInt(value,10);
+            if (isNaN(val)) {
+                return false;
+            }
+            if (!angular.isNumber(val) || val > $scope.maxGenesLimit || val <= 0 || val > $scope.genesLimit) {
```

Comment:
Again, the pattern is duplicated here. You are doing the same logic twice. This is has to do with the representation a bit, but I see how it may be difficult to untangle at this point.

I was getting at this

```js
const hasValidCount = (entity) => {
    const entityCount = parseInt(entity.count, 10);
    return !isNaN(entityCount) &&
        _.inRange(entityCount, 0, Math.min(entity.limit, Math.min(entity.maxLimit)+1);
}
```